v2/tools/asoctl/internal/crd/cleaner_test.go

```diff
@@ -0,0 +1,311 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package crd
+
+import (
+	"context"
+	"testing"
+
+	resources "github.com/Azure/azure-service-operator/v2/api/resources/v1alpha1api20200601"
+	"github.com/Azure/azure-service-operator/v2/internal/controllers"
+	"github.com/Azure/go-autorest/autorest/to"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/fake"
+	apiextensions "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	fake2 "sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+// TODO: Currently we need to create clientsets for each test as they run in parallel and we run into `resource already exists` error.
+// TODO: We may require a testing suite re-use the clientsets efficiently.
+func getClientSets() (apiextensions.ApiextensionsV1Interface, client.WithWatch, *Cleaner) {
```

Comment:
minor: consider calling this `makeClientSets` to match slightly closer to the languages terminology - we're not really "get"-ing something from somewhere in here we're constructing stuff, so either make or new make more sense IMO.